cmake/modules/BuildQuiche.cmake

```
@@ -0,0 +1,42 @@
+# build Quiche as a dependency of rgw_h3_quiche
+
+function(build_quiche)
+  if(NOT BORINGSSL_SOURCE_DIR)
+    message(FATAL_ERROR "BuildQuiche requires BORINGSSL_SOURCE_DIR from BuildBoringSSL")
+  endif()
+  list(APPEND quiche_DEPENDS boringssl_ext)
+
+  # build the library with cargo, but invoke it with cmake -E env
+  # to pass in the boringssl path
+  set(cargo_command ${CMAKE_COMMAND} -E env QUICHE_BSSL_PATH=${BORINGSSL_SOURCE_DIR} -- cargo build)
```

```suggestion
  set(cargo_command ${CMAKE_COMMAND} -E env QUICHE_BSSL_PATH=${BORINGSSL_SOURCE_DIR} cargo build)
```
resolves a compile error:
```
[89/950] Performing build step for 'quiche_ext'
FAILED: src/rgw/quiche_ext-prefix/src/quiche_ext-stamp/quiche_ext-build src/rgw/quiche/target/release/libquiche.a src/rgw/quiche/target/release/quiche.pc
cd /mnt/nvme0n1p1/src-git/QUIC/ceph--up--main/build/src/rgw/quiche && /usr/bin/cmake -E env QUICHE_BSSL_PATH=/mnt/nvme0n1p1/src-git/QUIC/ceph--up--main/buil
d/src/rgw/boringssl/src -- cargo build --features ffi,pkg-config-meta,qlog --release && /usr/bin/cmake -E touch /mnt/nvme0n1p1/src-git/QUIC/ceph--up--main/b
uild/src/rgw/quiche_ext-prefix/src/quiche_ext-stamp/quiche_ext-build
cmake -E env: unknown option '--'
```

on 
```
cat /etc/redhat-release
CentOS Stream release 8
```
cmake version:
```
dnf info cmake
Installed Packages
Name         : cmake
Version      : 3.20.2
Release      : 5.el8
Architecture : x86_64
Size         : 31 M
Source       : cmake-3.20.2-5.el8.src.rpm
Repository   : @System
From repo    : appstream
...
```

---

ah thanks. the docs at https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-e-arg-env say the double-dash is optional, and `New in version 3.24`. i'll remove those